### ISSUE TITLE
FlsFree not called, leading to crash

### ISSUE THREAD

**Issue by levinli303** [gnustep/libobjc2#321]
`FlsAlloc` is called here, but we are not calling `FlsFree` when DLL is unloaded. This leads to crash that is mentioned [here](https://learn.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/fatal-error-thread-exit-fls-callback).

https://github.com/gnustep/libobjc2/blob/0c1a89373f137489a829c69009d7ce752ac46b85/arc.mm#L57

I'm seeing this crash when objc.dll is used in a shell extension.

![Screenshot 2024-12-08 at 8 48 54 PM](https://github.com/user-attachments/assets/d89dcc04-6833-4171-97de-81056e06fab4)

<img width="754" alt="Screenshot 2024-12-08 at 8 50 31 PM" src="https://github.com/user-attachments/assets/f359e39d-1f32-4968-ad33-c866dd846a52">

**Comment by davidchisnall** [comment]
Unloading dynamic libraries containing Objective-C is unsupported and will crash for a great many reasons, but this looks as if you're unloading objc.dll?  That's *definitely* not supported and would require some significant changes to the library.

**Comment by levinli303** [comment]
> Unloading dynamic libraries containing Objective-C is unsupported and will crash for a great many reasons, but this looks as if you're unloading objc.dll? That's _definitely_ not supported and would require some significant changes to the library.

I don't have code to unload objc.dll. This might have come from explorer unloading my dll (it is a shell extension), and thus unloading the dlls that is no longer used. If that is the case, I could probably uninitialize the objc runtime when my dll is unloaded, but there does not seem to be a way to do that.

**Comment by davidchisnall** [comment]
No, because there is no good way of doing it. Pointers to internal library state flow everywhere and if we miss one then we introduce a security vulnerability. The risk in almost correctly supporting library unloading is far higher than the benefit from supporting it.
